lib/slave/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+
+target_sources(kickcat PRIVATE
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/AbstractESC.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/AbstractSPI.cc
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/ESC/Lan9252.cc
+)
+
+target_include_directories(kickcat PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+target_include_directories(kickcat PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include/kickcat)
+
+if (BUILD_DRIVER_NUTTX)
+  add_subdirectory(driver/nuttx)
```

Comment:
Seems overkill to add a CMakelists for one file. Just andle the condition directly here.

Reply:
or check for the `__NUTTX__` variable that is passed when building with nuttx

Reply:
I went for the solution to add a set(NUTTX 1) variable in the nuttx toolchain. I will propose a PR on the NUTTX project to add this behavior.

Reply:
> or check for the `__NUTTX__` variable that is passed when building with nuttx

This cannot be done at CMakeLists level: this is a preprocessor definition. SO yes it could be used but I prefer to avoid mixing sources to build in the targeted source code itself.